src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-param-reassign */
+
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+export const fetchUserPosts = createAsyncThunk<Post[], number>(
+  'posts/fetchUserPosts',
+  userId => {
+    const response = getUserPosts(userId);
```

Comment:
You should consider error handling for the async operation. If the `getUserPosts` call fails, you want to catch that error and handle it appropriately, for example by dispatching a rejected action with the error.